kingpin/actors/aws/ecs.py

```diff
@@ -0,0 +1,410 @@
+
+    def _handle_failures(self, failures):
+        """Logs ECS failures. Raises an unrecoverable exception if there are any.
+
+        Args:
+            failures: list of failures from ECS api.
+        """
+        if not failures:
+            return
+
+        # TODO: When can we recover? What are the failures? How do we recover?
+        for failure in failures:
+            self.log.error(failure)
+        raise exceptions.RecoverableActorFailure()
+
+
+class RunTask(ECSBaseActor):
+
+    """Register and run a task on ECS.
+
+    This actor will loop indefinitely until the task is complete.
+
+    **Options**
+
+    :region:
+      AWS region (or zone) name, such as us-west-2 or eu-west-1.
+
+    :cluster:
+      ECS cluster in the region to launch the task into.
+
+    :task_definition:
+      String of path to the Task Definition file template.
+      Must be a local file path.
+      Parameters to be interpolated must be of the form %VAR%.
+
+    :parameters:
+      A dictionary of key/value pairs used to fill in the parameters for the
+      Task Definition template. Default: {}.
+
+    :count:
+      How many tasks to run. Default: 1.
+
+    :wait:
+      Whether to wait for the tasks to complete. Default: True.
+
+    **Examples**
+
+    .. code-block:: yaml
+
+       actor: aws.ecs.RunTask
+       desc: Run migrations
+       options:
+          task_definition: nextdoor-migrate.json
+          region: us-west-2
+          cluster: us1-internal
+
+    **Dry Mode**
+
+    Will only attempt to interpolate env vars into the task definition.
+    """
+
+    all_options = {
+        'region': (str, REQUIRED,
+                   'AWS region (or zone) name, like us-west-2 or eu-west-1.'),
+        'cluster': (str, REQUIRED,
+                    'ECS cluster in the region to launch the task into.'),
+        'task_definition':
+            (str, REQUIRED,
+             'String of path to the Task Definition file template. '
+             'Must be a local file path. '
+             'Parameters to be interpolated must be of the form %VAR%.'),
+        'parameters': (dict, {},
+                       'A dictionary of key/value pairs '
+                       'used to fill in the parameters for the template.'),
+        'count': (int, 1, 'How many tasks to run.'),
+        'wait': (bool, True,
+                 'Whether to wait for the tasks to complete.')
+    }
+
+    @gen.coroutine
+    def _register_task(self, task_definition):
+        """Registers a task on ECS.
+
+        Args:
+            task_definition: dict of ECS task definition parameters.
+
+        Returns:
+            tuple: ('family', 'revision').
+        """
+        # Validate the schema.
+        try:
+            jsonschema.validate(task_definition, TASK_DEFINITION_SCHEMA)
+        except jsonschema.exceptions.ValidationError as e:
+            raise exceptions.InvalidOptions(e)
+
+        if self._dry:
+            self.log.warning('Would register task')
```

Comment:
Should be Info not Warning. Any information you could print here? maybe family and version?